docs/tutorial.md

```diff
@@ -0,0 +1,159 @@
+#  kpack Tutorial
+
+This tutorial will walk through creating a kpack [image](image.md) resource to build a docker image from source and allow kpack to keep the image upto date.  
+
+1. Make sure you have kpack installed on your cluster and have access via kubectl
+
+    Note: Follow these docs to install kpack. 
+
+1. Create a secret with push credentials for the docker registry that you plan on publishing images to with kpack.  
+
+    ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: registry-credentials
+      annotations:
+        build.pivotal.io/docker: <registry>
+    type: kubernetes.io/basic-auth
+    stringData:
+      username: <username>
+      password: <password>
+    ```
+   
+   Note: The secret must be annotated with the registry prefix for its corresponding registry. For [dockerhub](https://hub.docker.com/) this should be `index.docker.io`. 
+   For gcr this should be `gcr.io`.
+   
+   Your secret configuration should look something like this:
+   
+   ```yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: tutorial-registry-credentials
+      annotations:
+        build.pivotal.io/docker: index.docker.io
+    type: kubernetes.io/basic-auth
+    stringData:
+      username: sample-username
+      password: sample-password
+    ```
+   
+   Apply that credential to the cluster 
+   
+    ```bash
+   kubectl apply -f <name-of-secret-file.yaml>
+    ```
+   
+   Note: Learn more about kpack secrets with the [kpack secret documentation](secrets.md): 
+
+1. Create a service account that references the registry secret created above 
+
+     ```yaml
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: tutorial-service-account
+    secrets:
+      - name: tutorial-registry-credentials
+     ```
+    
+    Apply that service account to the cluster 
+   
+     ```
+     kubectl apply -f <name-of-service-account-file.yaml>
+     ```
+
+1. Apply kpack image configuration 
+
+    An image configuration is the specification for an image that kpack should build and manage. For more info check out image documentation. We will create a sample image that builds with the default builder setup in installing.      
+      
+    Create an image configuration:
+    
+    ```yaml
+    apiVersion: build.pivotal.io/v1alpha1
+    kind: Image
+    metadata:
+      name: tutorial-image
+    spec:
+      tag: <REGISTRY>/project-name/app
+      serviceAccount: tutorial-service-account
+      builder:
+        name: default-builder
+        kind: ClusterBuilder
+      source:
+        git:
+          url: <YOUR-GIT-URL>
+          revision: master
+    ```
+
+    - Make sure to replace <REGISTRY> with the registry you configured in step #2.     
+    - Make sure to replace <YOUR-GITHUB-URL> with a publicly accessible github url     
```

Comment:
Since this is a tutorial, I wonder if it would be easier to supply the github.com/buildpack/sample-java-app here instead. That way, we know they're using an app that has a supported buildpack.